src/Connection/Http1Connection.php

```diff
@@ -374,24 +382,33 @@ private function readResponse(
                             // to resolve promise with headers.
                             $chunk = null;
 
-                            /** @psalm-suppress PossiblyNullReference */
-                            do {
-                                /** @noinspection CallableParameterUseCaseInTypeContextInspection */
-                                $parser->parse($chunk);
-                                /** @noinspection NotOptimalIfConditionsInspection */
-                                if ($parser->isComplete()) {
-                                    break;
-                                }
-
-                                if (!$backpressure instanceof Success) {
-                                    yield $this->withCancellation($backpressure, $bodyCancellationToken);
-                                }
-
-                                /** @psalm-suppress TypeDoesNotContainNull */
-                                if ($this->socket === null) {
-                                    throw new SocketException('Socket closed prior to response completion');
-                                }
-                            } while (null !== $chunk = yield $this->socket->read());
+                            try {
+                                /** @psalm-suppress PossiblyNullReference */
+                                do {
+                                    /** @noinspection CallableParameterUseCaseInTypeContextInspection */
+                                    $parser->parse($chunk);
+                                    /** @noinspection NotOptimalIfConditionsInspection */
+                                    if ($parser->isComplete()) {
+                                        break;
+                                    }
+
+                                    /** @psalm-suppress TypeDoesNotContainNull */
```

Comment:
I guess the suppression here is also a rebase artifact from the `if` below?

Reply:
Yeah, funny that I only got that line.